bika/lims/browser/calcs.py

```diff
@@ -258,17 +255,18 @@ def calculate(self, uid=None):
                 else:
                     self.alerts[uid] = [alert, ]
 
-        if analysis.portal_type == 'ReferenceAnalysis':
-            # The analysis is a Control or Blank. We might use the
-            # reference results instead other specs
-            _uid = analysis.getServiceUID()
-            specs = analysis.aq_parent.getResultsRangeDict().get(_uid, {})
-
-        else:
-            # Get the specs directly from the analysis. The getResultsRange
-            # function already takes care about which are the specs to be used:
-            # AR, client or lab.
-            specs = analysis.getResultsRange()
+        # XXX ramonski-2017-12-08: 'specs' is nowhere used, so I'm disabling this block
+        #
+        # if analysis.portal_type == 'ReferenceAnalysis':
+        #     # The analysis is a Control or Blank. We might use the
+        #     # reference results instead other specs
+        #     _uid = analysis.getServiceUID()
+        #     specs = analysis.aq_parent.getResultsRangeDict().get(_uid, {})
+        # else:
+        #     # Get the specs directly from the analysis. The getResultsRange
+        #     # function already takes care about which are the specs to be used:
+        #     # AR, client or lab.
+        #     specs = analysis.getResultsRange()
```

Comment:
Yes, you are right, this block of code is no longer used since we introduced `format_numeric_result` in L273 as per https://github.com/senaite/bika.lims/commit/d553f0c5f4b6fdb829a82ae699897c180b751f97#diff-bddafb05eeff20ad5a33aab50565cba2 . You can safely remove this block instead of commenting it. 

As an adiditional note, the specs for each analysis are obtained during the analyses list rendering in [`bika.lims.browser.analyses.py#L758-L770`](https://github.com/senaite/bika.lims/blob/master/bika/lims/browser/analyses.py#L758-L770).

In any case, seems evident to me this whole class needs to be refactored asap.